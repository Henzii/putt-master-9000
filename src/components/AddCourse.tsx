import React, { useEffect, useState } from "react";
import { View, StyleSheet, Text } from "react-native";
import { ActivityIndicator, Button, Caption, Headline, Subheading, Switch, TextInput } from "react-native-paper";
import useGPS from "../hooks/useGPS";
import Divider from "./ThemedComponents/Divider";
import { Coordinates, Course } from "../types/course";
import { GPShookReturn } from "../types/gps";
import SelectLocationFromMap from "./SelectLocationFromMap";

type AddCourseProps = {
    onCancel?: () => void,
    onAdd?: (name: string, coordinates: Coordinates, courseId?: string) => void,
    loading?: boolean,
    course?: Course
}

const AddCourse = ({ onCancel, onAdd, course, loading = false }: AddCourseProps) => {
    const [newName, setNewName] = useState(course?.name ?? '');
    const [manualCoordinates, setManualCoordinates] = useState(!!course);
    const [displayMap, setDisplayMap] = useState(false);
    const [lat, setLat] = useState<string | undefined>(course?.location.coordinates[1].toString());
    const [lon, setLon] = useState<string | undefined>(course?.location.coordinates[0].toString());
    const gps = useGPS();
    useEffect(() => {
        if (gps.ready && !manualCoordinates) {
            setLat(gps.lat?.toString());
            setLon(gps.lon?.toString());
        }
    }, [gps]);
    const handleAddCourse = () => {
        const coords = {
            lat: Number.parseFloat(lat || '0') || 0,
            lon: Number.parseFloat(lon || '0') || 0,
        };
        if (onAdd) onAdd(newName, coords, course?.id.toString());
    };

    if (displayMap) {
        return (
            <SelectLocationFromMap
                onDismiss={() => setDisplayMap(false)}
                initialRegion={{ latitude: Number(lat) || 0, longitude: Number(lon) || 0, latitudeDelta: 0.5, longitudeDelta: 0.5 }}
                onSelect={((region) => {
                    setLat(region.latitude.toFixed(7));
                    setLon(region.longitude.toFixed(7));
                    setDisplayMap(false);
                })}
            />
        );
    }

    return (
        <View style={tyyli.root}>
            <Headline testID="AddCourseTitle">{course ? 'Edit' : 'Add'} Course</Headline>
            <Subheading>Name</Subheading>
            <TextInput value={newName} autoComplete='off' mode="outlined" label="Course name" onChangeText={(value) => setNewName(value)} />
            <Divider />
            <LocationForm
                latitude={lat}
                longitude={lon}
                gps={gps}
                setLat={setLat}
                setLon={setLon}
                manualCoordinates={manualCoordinates}
                setManualCoordinates={setManualCoordinates}
            />
            {gps.error && <Caption style={{ color: 'red' }}>{gps.error} Set coordinates manually</Caption>}
            <Button disabled={!manualCoordinates} onPress={() => setDisplayMap(true)}>Select location from map</Button>
            <Divider />
            <View style={[tyyli.split, { margin: 20 }]}>
                <Button icon="check" onPress={handleAddCourse} mode="contained" loading={loading} disabled={loading}>
                    {course ? 'Save' : 'Add'}
                </Button>
                <Button icon="cancel" onPress={onCancel} mode="outlined">Cancel</Button>
            </View>
        </View>
    );

};
const LocationForm = ({ latitude, longitude, setLat, setLon, gps, manualCoordinates, setManualCoordinates }: {
    latitude?: string, longitude?: string,
    setLat: (v: string) => void, setLon: (v: string) => void,
    gps?: GPShookReturn,
    manualCoordinates: boolean
    setManualCoordinates: (v: boolean) => void
}) => {
    return (
        <>
            <Subheading>Location</Subheading>
            <View style={[tyyli.split, { justifyContent: 'space-between' }]}>
                <Text>Set coordinates manually</Text>
                <Switch value={manualCoordinates} onChange={() => setManualCoordinates(!manualCoordinates)} />
            </View>
            <View style={tyyli.split}>
                <TextInput
                    value={latitude?.toString()}
                    disabled={!manualCoordinates}
                    autoComplete='off'
                    label="Lat"
                    mode="outlined"
                    style={{ minWidth: 120 }}
                    keyboardType="numeric"
                    onChangeText={setLat}
                />
                <TextInput
                    value={longitude?.toString()}
                    disabled={!manualCoordinates}
                    autoComplete='off'
                    label="Long"
                    mode="outlined"
                    style={{ minWidth: 120 }}
                    keyboardType="numeric"
                    onChangeText={setLon}
                />
            </View>
            <Text style={{ marginTop: 10 }}>GPS accruacy: {Math.floor(gps?.acc || 0)}m</Text>
            {gps?.loading && (
                <View style={{ flexDirection: 'row', gap: 5 }}>
                    <ActivityIndicator size="small" />
                    <Text>GPS Loading...</Text>
                </View>
            )}
        </>
    );
};
const tyyli = StyleSheet.create({
    root: {
        width: '90%',
        height: '90%',
        backgroundColor: '#fafafa',
        padding: 20,
    },
    split: {
        display: 'flex',
        justifyContent: 'space-evenly',
        alignItems: 'center',
        flexDirection: 'row',
    }
});

export default AddCourse;
