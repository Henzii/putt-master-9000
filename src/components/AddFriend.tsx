import React from 'react';
import { View, StyleSheet, ScrollView } from "react-native";
import { ActivityIndicator, IconButton, Text, TextInput, useTheme } from 'react-native-paper';
import { SEARCH_USER } from '../graphql/queries';
import useTextInput from '../hooks/useTextInput';
import { useLazyQuery } from '@apollo/client';
import { SafeUser } from '../types/user';
import Stack from './Stack';
import Spacer from './ThemedComponents/Spacer';
import { MD3Colors } from 'react-native-paper/lib/typescript/types';
import { useSession } from '@hooks/useSession';

type AddFriendProps = {
    onClose?: () => void,
    onAddFriend: (friendName: string) => void,
    friends: SafeUser[]
}

type QueryResponse = {
    searchUser: {
        users: SafeUser[]
    }
}

const AddFriend = ({ onClose, onAddFriend, friends }: AddFriendProps) => {
    const [searchUsers, { data, loading }] = useLazyQuery<QueryResponse>(SEARCH_USER);
    const { colors } = useTheme();
    const { id } = useSession();
    const searchTextInput = useTextInput({ callBackDelay: 1000, defaultValue: '' }, (value) => {
        searchUsers({ variables: { search: value.toLowerCase() } });
    });

    const tyyli = createStyles(colors);

    const handleAddFriend = (name: string) => {
        if (!name.trim()) {
            return;
        }
        onAddFriend(name);
        if (onClose) onClose();
    };

    const users = data?.searchUser?.users.slice(0, 50) ?? [];
    return (
        <View style={tyyli.container}>
            <Stack direction="row" justifyContent="space-between" alignItems='center'>
                <Text variant="headlineSmall">Add friend</Text>
                <IconButton icon="close" onPress={onClose} />
            </Stack>
            <Text variant="titleSmall">Who&apos;s your daddy</Text>
            <TextInput
                autoComplete='off'
                {...searchTextInput}
                mode='outlined'
                label='Name'
            />
            {loading ? <ActivityIndicator /> : users.length > 0 && <Text>{users.length} users found</Text>}
            <Spacer />
            <ScrollView>
                <Stack gap={8}>
                    {users?.map(user => {
                        const isFriend = friends.some(f => f.id === user.id);
                        const disabled = user.blockFriendRequests || isFriend || user.id === id;
                        return (
                            <View key={user.id} style={tyyli.user}>
                                <Text variant="labelLarge" style={tyyli.name}>{user.name}</Text>
                                {isFriend && <Text style={tyyli.badge}>Friend!</Text>}
                                {user.id === id && <Text style={tyyli.badge}>It&apos;s a me!</Text>}
                                {!isFriend && user.id !== id && (
                                    <IconButton
                                        icon="account-plus"
                                        iconColor={colors.primary}
                                        containerColor={disabled ? colors.surfaceDisabled : colors.tertiary}
                                        disabled={disabled}
                                        onPress={() => handleAddFriend(user.name)}
                                    />
                                )}
                            </View>
                        );
                    })}
                </Stack>
            </ScrollView>
        </View>
    );
};

const createStyles = (color: MD3Colors) => StyleSheet.create({
    container: {
        padding: 10,
        paddingTop: 20,
        flex: 1,
        maxHeight: '100%'
    },
    badge: {
        color: color.primary,
        borderColor: color.primary,
        borderWidth: 1,
        fontWeight: 'bold',
        padding: 5,
        borderRadius: 10

    },
    user: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingVertical: 5,
        paddingHorizontal: 10,
        borderWidth: 1,
        borderColor: color.elevation.level4,
        minHeight: 80,
        gap: 5
    },
    name: {
        flexShrink: 1
    }
});

export default AddFriend;