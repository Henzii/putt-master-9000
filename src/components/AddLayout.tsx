import React, { useState } from 'react';
import { StyleSheet, ScrollView, View, Text } from "react-native"
import { Button, Caption, Divider, Headline, TextInput, Title } from "react-native-paper"
import { Layout } from '../hooks/useCourses';

const AddLayout = ({ onCancel, onAdd }: AddLayoutProps) => {
    const [holes, setHoles] = useState<number | null>(null)
    const [pars, setPars] = useState<number[]>([])
    const [name, setName] = useState('')

    const handleHolesChange = (value: string) => {
        const holesInt = Number.parseInt(value);
        if (!isNaN(holesInt)) {
            setHoles(holesInt)
            setPars(Array(holesInt).fill(3))
        }
    }
    const handleParChange = (hole: number, rawPar: number | string) => {
        let par = rawPar;
        if (typeof par === 'string') {
            par = Number.parseInt(par)
        }
        if (isNaN(par)) return;
        
        const parsCopy = [...pars]
        parsCopy[(hole-1)] = par;
        setPars(parsCopy)
    }
    console.log(pars)
    const handleCancel = () => {
        if (onCancel) onCancel()
    }
    const handleAdd = () => {
        const newLayout: Omit<Layout, 'id'> = {
            name,
            pars,
            par: pars.reduce((p,c) => p+c, 0),
            holes: holes || 0,
        }
        if (onAdd) onAdd(newLayout)
    }
    return (
        <ScrollView style={tyyli.main} contentContainerStyle={{ paddingBottom: 50, }}>
            <Headline>Add layout</Headline>
            <Caption>Layout name</Caption>
            <TextInput autoComplete={false} value={name} onChangeText={(value) => setName(value)}/>
            <Caption>Number of holes</Caption>
            <TextInput autoComplete={false} keyboardType='numeric' onChangeText={handleHolesChange} />
            <Divider />
            <Title>Pars</Title>
            <HolesPars holes={holes} onParChange={handleParChange}/>
            <Text>
                {holes || 0} Holes, par {pars.reduce((p,c) => p+c,0)}
            </Text>
            <View style={tyyli.buttonsContainer}>
                <Button onPress={handleAdd}>Add</Button>
                <Button onPress={handleCancel}>Cancel</Button>
            </View>
        </ScrollView>
    )
}

const HolesPars = ({ holes, onParChange }: { holes: number | null, onParChange: (hole: number, par: number | string) => void }) => {
    if (!holes) return null;
    const returni = []
    for (let i = 1; i <= holes; i++) {
        returni.push(<TextInput
            style={tyyli.singlePar}
            onChangeText={(value) => onParChange(i, value)}
            mode='outlined'
            label={`Hole ${i}`}
            defaultValue='3'
            dense
            autoComplete={false}
            keyboardType='numeric'
        />)
    }
    return (
        <View style={tyyli.parList}>
            {returni}
        </View>
    )
}

const tyyli = StyleSheet.create({
    main: {
        backgroundColor: '#fff',
        height: '90%',
        width: '90%',
        padding: 20,
    },
    singlePar: {
        maxWidth: 100,
        margin: 10,
    },
    parList: {
        marginBottom: 10,
    },
    buttonsContainer: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-evenly'
    }
})
type AddLayoutProps = {
    onCancel?: () => void,
    onAdd?: (layout: Omit<Layout, "id">) => void,
}

export default AddLayout;