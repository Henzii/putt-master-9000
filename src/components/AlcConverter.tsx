import React, { useMemo, useState } from "react";
import { StyleSheet, View, Text } from "react-native";
import { Headline, TextInput } from "react-native-paper";
import Container from "./ThemedComponents/Container";

type DrinkInputProps = {
    drinkName: string,
    handleChange: (d: Drink, val: string) => void,
    testID?: string,
}
function DrinkInput({ drinkName, handleChange, testID }: DrinkInputProps) {
    return (
        <TextInput
            testID={testID}
            style={styles.input}
            autoComplete={false}
            mode="outlined"
            dense
            onChangeText={(val) => handleChange(drinkName as Drink, val)}
            keyboardType='numeric'
        />
    );
}
type Drink = 'long' | 'wine' | 'strong' | 'normal' | 'longstrong'

export const defaultValues = {
    strong: {
        count: 0,
        multiplier: 1.25,
    },
    longstrong: {
        count: 0,
        multiplier: 1.9,
    },
    wine: {
        count: 0,   // dl
        multiplier: 0.9,
    },
    long: {
        count: 0,
        multiplier: 1.5151,
    },
    normal: {
        count: 0,
        multiplier: 1,
    }
};

export default function AlcConverter() {
    const [drinks, setDrinks] = useState(defaultValues);
    const beers = useMemo(() => {
        const sum = Object.values(drinks).reduce((prev: number, curr) => {
            return prev + (curr.count * curr.multiplier);
        }, 0);
        return Math.round(sum*100)/100;
    }, [drinks]);
    const handleChange = (drink: Drink, val: string) => {
        const valInt = Number.parseInt(val) || 0;
        if (isNaN(valInt)) return;
        setDrinks({ ...drinks, [drink]: { ...drinks[drink], count: valInt } });
    };
    return (
        <Container noPadding withScrollView>
            <Headline>Beerculator</Headline>
            <View style={styles.row}>
                <Text style={styles.text}>Long (0,5l)</Text>
                <DrinkInput drinkName="long" handleChange={handleChange} testID="long" />
            </View>
            <View style={styles.row}>
                <Text style={styles.text}>Strong (&gt;5.0%)</Text>
                <DrinkInput drinkName="strong" handleChange={handleChange} testID="strong" />
            </View>
            <View style={styles.row}>
                <Text style={styles.text}>Long &amp; Strong</Text>
                <DrinkInput drinkName="longstrong" handleChange={handleChange} testID="longandstrong" />
            </View>
            <View style={styles.row}>
                <Text style={styles.text}>Wine (dl)</Text>
                <DrinkInput drinkName="wine" handleChange={handleChange} testID="wine" />
            </View>
            <View style={styles.row}>
                <Text style={styles.text}>Normal (0,33l)</Text>
                <DrinkInput drinkName="normal" handleChange={handleChange} testID="normal" />
            </View>
            <View style={[{ marginTop: 20 }, styles.row]}>
                <Text style={styles.text}>Total</Text>
                <Text testID="beers">{beers} beers</Text>
            </View>

        </Container>
    );
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        flexDirection: 'row',
    },
    row: {
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
    },
    text: {
        minWidth: 120,
    },
    input: {
        width: 100,
    }
});
