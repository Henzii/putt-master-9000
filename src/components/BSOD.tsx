import React, { useEffect, useState } from 'react';
import { View, Text, StyleSheet } from "react-native";
import * as ExpoUpdates from 'expo-updates';

const BSOD = () => {
    const [counter, setCounter] = useState(0);

    useEffect(() => {
        if (counter >= 10) {
            ExpoUpdates.reloadAsync();
        }
    }, [counter]);

    useEffect(() => {
        const interval = setInterval(() => {
            setCounter(val => val + 1);
        }, 1500);

        return () => clearInterval(interval);
    }, []);
    return (
        <View style={styles.container}>
            <Text style={styles.smiley}>:(</Text>
            <Text style={styles.text}>Your FuDisc ran into a problem and needs to restart. We&apos;re faking
                some error info collection, and then we&apos;ll restart for you.
            </Text>
            <Text style={styles.text}>{counter}0% complete</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#0078d7',
        height: '100%',
        padding: 20
    },
    smiley: {
        color: 'white',
        fontSize: 80
    },
    text: {
        marginTop: 15,
        color: 'white',
        fontSize: 18,
        fontWeight: '200',
    }
});

export default BSOD;