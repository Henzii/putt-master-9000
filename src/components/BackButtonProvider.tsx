import React, { createContext, useContext, useState } from "react";
import { useNavigate } from "react-router-native";

const BackContext = createContext<BackActions>({} as BackActions);

export const useBackButton = () => useContext(BackContext);

interface BackActions {
    setDestination: (dest: string) => void,
    setCallBack: (cb: () => void) => void,
    goBack: () => void,
}

export default function BackButtonProvider ( {children}: { children: React.ReactElement | Element | false | null }) {
    const [destination, setDestination] = useState<string | undefined>();
    const [callBack, setStateCallBack] = useState<() => void | undefined>();
    const navigate = useNavigate();

    const backActions = {
        setDestination: (dest: string | undefined) => {
            setDestination(dest);
        },
        setCallBack: (cb: () => void | undefined) => {
            setStateCallBack(() => cb);
        },
        goBack: () => {
            if (callBack) {
                callBack();
                setStateCallBack(undefined);
            } else if (destination) {
                navigate(destination);
                setDestination(undefined);
            } else {
                navigate(-1);
            }
            return true;
        }
    };
    return (
        <BackContext.Provider value={backActions}>
            {children}
        </BackContext.Provider>
    );
}