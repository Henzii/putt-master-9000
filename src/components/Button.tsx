import React, { useState } from 'react';
import { View, Text, Pressable, StyleSheet, TouchableHighlightComponent } from 'react-native';
import { theme } from '../utils/theme';

export default function Button ({ text, onClick, width, heigth, ...props }:ButtonProps) {
    const [pressed, setPressed] = useState(false);
    const tyyli = StyleSheet.create({
        main: {
            backgroundColor: props.backgroundColor || theme.colors.primary,
            width: (props.fullWidth ? '100%' : (width || 'auto')),
            height: heigth || '120%',
            padding: (props.fullWidth ? 20 : 5),
            borderRadius: (props.fullWidth ? 0 : 10),
            borderColor: props.borderColor || 'black',
            borderWidth: props.borderWidth || 1,
            shadowColor: '#000',
            shadowOpacity: props.shadowOpacity || 0.5,
            shadowRadius: 4,
            shadowOffset: props.shadowOffcet || { width: 4, height: 4},
        },
        pressed: {
            shadowOffset: { width: -1, height: -1},
        },
        text: {
            textAlign: 'center',
            fontSize: theme.font.sizes.large,
            color: (props.color ? theme.font.color[props.color] : theme.font.color.primary)
        }
    });
    return (
        <View style={[tyyli.main, (pressed) ? tyyli.pressed: null ]}>
            <Pressable onPress={onClick} onPressIn={() => setPressed(true)} onPressOut={() => setPressed(false)}>
                <Text style={tyyli.text}>
                    {text}
                </Text>
            </Pressable>
        </View>
    );
}

export type ButtonProps = {
    text: string,
    onClick?: () => void,
    color?: 'primary' | 'secondary',
    width?: string,
    heigth?: number,
    fullWidth?: boolean
    borderColor?: string,
    borderWidth?: number,
    backgroundColor?: string,
    shadowOpacity?: number,
    shadowOffcet?: { width: number, height: number }
}
