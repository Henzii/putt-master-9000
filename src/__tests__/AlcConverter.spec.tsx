import React from 'react';
import AlcConverter from '../components/AlcConverter';
import { render, fireEvent } from '@testing-library/react-native';

import { defaultValues as dv } from '../components/AlcConverter';

describe('<AlcConverter /> testit', () => {
    it('RenderÃ¶ityy oikein...', () => {
        const puu = render(<AlcConverter />).toJSON();
        expect(puu).toMatchSnapshot();
    });
    it('Laskuri toimii...', () => {
        const { getByTestId } = render(<AlcConverter />);
        const result = getByTestId('beers');
        // Alussa 0
        expect(result.children.join('')).toContain('0 beers');

        // Asetetaan eri arvoja kenttiin
        fireEvent.changeText(getByTestId('long'), '2');
        fireEvent.changeText(getByTestId('strong'), '1');
        fireEvent.changeText(getByTestId('longandstrong'), '3');
        fireEvent.changeText(getByTestId('wine'), '4');
        fireEvent.changeText(getByTestId('normal'), '6');
        // Lasketaan summa em juomille
        let summa =   dv.long.multiplier * 2 +
                        dv.strong.multiplier * 1 +
                        dv.longstrong.multiplier * 3 +
                        dv.wine.multiplier * 4 +
                        dv.normal.multiplier * 6;
        summa = Math.round(summa*100)/100;
        expect(result.children[0]).toContain(summa);
    });
});