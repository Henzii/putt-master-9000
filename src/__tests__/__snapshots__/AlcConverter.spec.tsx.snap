// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlcConverter /> testit Render√∂ityy oikein... 1`] = `
<RCTScrollView
  contentContainerStyle={
    Array [
      undefined,
      undefined,
      undefined,
      false,
      Object {
        "backgroundColor": "#f6f6f6",
        "padding": 0,
      },
      undefined,
      undefined,
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 24,
                "letterSpacing": 0,
                "lineHeight": 32,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Beerculator
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "minWidth": 120,
          }
        }
      >
        Long (0,5l)
      </Text>
      <View
        style={
          Object {
            "width": 100,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 48,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <TextInput
              autoComplete="off"
              editable={true}
              keyboardType="numeric"
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#6200ee"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 40,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontVariant": undefined,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="long"
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "minWidth": 120,
          }
        }
      >
        Strong (&gt;5.0%)
      </Text>
      <View
        style={
          Object {
            "width": 100,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 48,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <TextInput
              autoComplete="off"
              editable={true}
              keyboardType="numeric"
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#6200ee"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 40,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontVariant": undefined,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="strong"
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "minWidth": 120,
          }
        }
      >
        Long & Strong
      </Text>
      <View
        style={
          Object {
            "width": 100,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 48,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <TextInput
              autoComplete="off"
              editable={true}
              keyboardType="numeric"
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#6200ee"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 40,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontVariant": undefined,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="longandstrong"
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "minWidth": 120,
          }
        }
      >
        Wine (dl)
      </Text>
      <View
        style={
          Object {
            "width": 100,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 48,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <TextInput
              autoComplete="off"
              editable={true}
              keyboardType="numeric"
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#6200ee"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 40,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontVariant": undefined,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="wine"
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "minWidth": 120,
          }
        }
      >
        Normal (0,33l)
      </Text>
      <View
        style={
          Object {
            "width": 100,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 48,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <TextInput
              autoComplete="off"
              editable={true}
              keyboardType="numeric"
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#6200ee"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 40,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontVariant": undefined,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="normal"
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "minWidth": 120,
          }
        }
      >
        Total
      </Text>
      <Text
        testID="beers"
      >
        0
         beers
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
